lib/bitcoin.py

```diff
@@ -754,7 +780,299 @@ def decrypt_message(self, encrypted):
+        ephemeral_pubkey = ephemeral.get_public_key_bytes(compressed=True)
+        encrypted = magic + ephemeral_pubkey + ciphertext
+        mac = hmac_oneshot(key_m, encrypted, hashlib.sha256)
+
+        return base64.b64encode(encrypted + mac)
+
+    @classmethod
+    def order(cls):
+        return CURVE_ORDER
+
+    def is_at_infinity(self):
+        return self == point_at_infinity()
+
+    @classmethod
+    def is_pubkey_bytes(cls, b: bytes):
+        try:
+            ECPubkey(b)
+            return True
+        except:
+            return False
+
+def generator():
+    return ECPubkey.from_point(generator_secp256k1)
+
+
+def point_at_infinity():
+    return ECPubkey(None)
+
+
+def sig_string_from_der_sig(der_sig: bytes, order=CURVE_ORDER) -> bytes:
+    r, s = ecdsa.util.sigdecode_der(der_sig, order)
+    return ecdsa.util.sigencode_string(r, s, order)
+
+
+def der_sig_from_sig_string(sig_string: bytes, order=CURVE_ORDER) -> bytes:
+    r, s = ecdsa.util.sigdecode_string(sig_string, order)
+    return ecdsa.util.sigencode_der_canonize(r, s, order)
+
+
+def der_sig_from_r_and_s(r: int, s: int, order=CURVE_ORDER) -> bytes:
+    return ecdsa.util.sigencode_der_canonize(r, s, order)
+
+
+def get_r_and_s_from_der_sig(der_sig: bytes, order=CURVE_ORDER) -> Tuple[int, int]:
+    r, s = ecdsa.util.sigdecode_der(der_sig, order)
+    return r, s
+
+
+def get_r_and_s_from_sig_string(sig_string: bytes, order=CURVE_ORDER) -> Tuple[int, int]:
+    r, s = ecdsa.util.sigdecode_string(sig_string, order)
+    return r, s
+
+
+def sig_string_from_r_and_s(r: int, s: int, order=CURVE_ORDER) -> bytes:
+    return ecdsa.util.sigencode_string_canonize(r, s, order)
+
+
+def is_secret_within_curve_range(secret: Union[int, bytes]) -> bool:
+    if isinstance(secret, bytes):
+        secret = string_to_number(secret)
+    return 0 < secret < CURVE_ORDER
+
+
+def construct_sig65(sig_string: bytes, recid: int, is_compressed: bool) -> bytes:
+    comp = 4 if is_compressed else 0
+    return bytes([27 + recid + comp]) + sig_string
+
+
+class ECPrivkey(ECPubkey):
+
+    def __init__(self, privkey_bytes: bytes):
+        assert_bytes(privkey_bytes)
+        if len(privkey_bytes) != 32:
+            raise Exception('unexpected size for secret. should be 32 bytes, not {}'.format(len(privkey_bytes)))
+        secret = string_to_number(privkey_bytes)
+        if not is_secret_within_curve_range(secret):
+            raise InvalidECPointException('Invalid secret scalar (not within curve order)')
+        self.secret_scalar = secret
+
+        point = generator_secp256k1 * secret
+        super().__init__(point_to_ser(point))
+        self._privkey = ecdsa.ecdsa.Private_key(self._pubkey, secret)
+
+    @classmethod
+    def from_secret_scalar(cls, secret_scalar: int):
+        secret_bytes = number_to_string(secret_scalar, CURVE_ORDER)
+        return ECPrivkey(secret_bytes)
+
+    @classmethod
+    def from_arbitrary_size_secret(cls, privkey_bytes: bytes):
```

Comment:
if this is already deprecated and unused, no point in bringing it in now ...

Reply:
Ok, done.